### PR TITLE
[v6] Fix undefined constant GLOB_BRACE on alpine

### LINKED ISSUE #1369: 6.16.2 version  GLOB_BRACE Constant error in Alpine system
> **warning:  Backup failed because: Use of undefined constant GLOB_BRACE - assumed 'GLOB_BRACE' (this will throw an Error in a future version of PHP).**

file path : https://github.com/spatie/laravel-backup/blob/6.16.2/src/Tasks/Backup/FileSelection.php

``` php
protected function sanitize($paths): Collection
    {
        return collect($paths)
            ->reject(function ($path) {
                return $path === '';
            })
            ->flatMap(function ($path) {
                return glob(str_replace('*', '{.[!.],}*', $path), GLOB_BRACE);
            })
            ->map(function ($path) {
                return realpath($path);
            })
            ->reject(function ($path) {
                return $path === false;
            });
    }
```

### PR DESCRIPTION
**Pull request by erikn69**
Closes #1369

### CODE REVIEW

File: src/Tasks/Backup/FileSelection.php [hunk 1]
```diff
@@ -159,7 +159,7 @@ protected function sanitize($paths): Collection
                 return $path === '';
             })
             ->flatMap(function ($path) {
-                return glob(str_replace('*', '{.[!.],}*', $path), GLOB_BRACE);
+                return strpos($path, '*') !== false && defined('GLOB_BRACE') ? glob(str_replace('*', '{.[!.],}*', $path), GLOB_BRACE) :  glob($path);
```
Comment (freekmurze) [inline src/Tasks/Backup/FileSelection.php:162]: Could you make this more readable by moving that conditional statement to a function that returns a boolean. Give that function a nice name.

**Review by freekmurze — CHANGES_REQUESTED** [review]

File: src/Tasks/Backup/FileSelection.php (hunk 1 above)
Comment (erikn69) [reply src/Tasks/Backup/FileSelection.php:162]: maybe more readable if a function return all the array

File: src/Tasks/Backup/FileSelection.php [hunk 2]
```diff
@@ -168,4 +168,20 @@ protected function sanitize($paths): Collection
...
+    /**
+     * @param string $path
+     *
+     * @return array
+     */
+    protected function getMatchingPaths(string $path): array
+    {
+        if (strpos($path, '*') !== false && defined('GLOB_BRACE')) {
+            $result = glob(str_replace('*', '{.[!.],}*', $path), GLOB_BRACE);
+        } else {
```
Comment (freekmurze) [inline src/Tasks/Backup/FileSelection.php:181]: Please refactor this else away by using early returns

File: src/Tasks/Backup/FileSelection.php [hunk 3]
```diff
@@ -168,4 +168,20 @@ protected function sanitize($paths): Collection
...
     }
+
+    /**
+     * @param string $path
+     *
+     * @return array
+     */
+    protected function getMatchingPaths(string $path): array
+    {
+        if (strpos($path, '*') !== false && defined('GLOB_BRACE')) {
```
Comment (freekmurze) [inline src/Tasks/Backup/FileSelection.php:179]: What does this `if` statement do if you would explain it to a human? Refactor that if statement to a method of it's own with a good explanatory name.

**Review by freekmurze — CHANGES_REQUESTED** [review]

File: src/Tasks/Backup/FileSelection.php (hunk 2 above)
Comment (erikn69) [reply src/Tasks/Backup/FileSelection.php]: else removed

File: src/Tasks/Backup/FileSelection.php (hunk 3 above)
Comment (erikn69) [reply src/Tasks/Backup/FileSelection.php]:
```php
protected function getMatchingPaths(string $path): array
{
     if (! defined('GLOB_BRACE')) {
         return glob($path);
     }
     return glob(str_replace('*', '{.[!.],}*', $path), GLOB_BRACE);
}
```
Better?? 
I'm sorry, maybe I have problems with my english

Comment (sbrow) [reply src/Tasks/Backup/FileSelection.php]: I believe he wants you to split it into 2 functions:

```php
    /**
     * @param string $path
     *
     * @return array
     */
    protected function getMatchingPaths(string $path): array
    {
        if ($this->canUseGlobBrace($path)) {
            return glob(str_replace('*', '{.[!.],}*', $path), GLOB_BRACE);
        }

        return glob($path);
    }

    protected function canUseGlobBrace(string $path): bool {
         return strpos($path, '*') !== false && defined('GLOB_BRACE');
    }
```

Comment (erikn69) [reply src/Tasks/Backup/FileSelection.php]: Thanks @sbrow

Comment (freekmurze) [reply src/Tasks/Backup/FileSelection.php]: Seems much better now thanks!

Is it possible to also add a test for this?

Comment (erikn69) [reply src/Tasks/Backup/FileSelection.php]: I don't know how to remove constant `GLOB_BRACE` for a test

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**